pkg_building.Rmd

```diff
@@ -255,11 +255,18 @@ Please do not list editors as contributors. Your participation in and contributi
 
 * Use `Imports` instead of `Depends` for packages providing functions from other packages. Make sure to list packages used for testing (`testthat`), and documentation (`knitr`, `roxygen2`) in your `Suggests` section of package dependencies (if you use `usethis` for adding testing infrastructure via [`usethis::use_testthat()`](https://usethis.r-lib.org/reference/use_testthat.html) or a vignette via [usethis::use_vignette()](https://usethis.r-lib.org/reference/use_vignette.html), the necessary packages will be added to `DESCRIPTION`). If you use any package in the examples or tests of your package, make sure to list it in `Suggests`, if not already listed in `Imports`.
 
+* If your (not Bioconductor) package depends on Bioconductor packages, make sure the installation instructions in the README and vignette are clear enough.
+    
+    * Is the automatic installation of Bioconductor packages by `install.packages()` enough? In that case, mention the user needs to run `setRepositories()` if they haven't set the necessary Bioconductor repositories yet. 
```

Comment:
I think that for binary packages, including `biocViews:` field in `DESCRIPTION` is enough. It is mainly when packages need to be installed via `BiocManager::install(.., type="source")` that this doesn't seem to work.

Reply:
Also, this only appears to be a problem when building the package under review from source code (either locally or from GitHub)

Reply:
I think biocViews doesn't do anything?

Reply:
My understanding form the thread on this issue https://github.com/r-lib/remotes/issues/380 (and other docs) is that it does a few things, but one is to indicate that bioconductor is a remote for one or more of the dependencies. For available binary dependencies that seems to be enough (it was in the case of acouple of the missing `BseSet` dependencies) but not for those requiring build from source.

Reply:
I'm not sure that BiocViews is needed for CRAN packages, maybe only for `remotes`  (usually used on CI tools) to catch a dependency on Bioconductor, but I don't know and it might be best to raise the question to R-devel or other sites.

I didn't know about `setRepositories`, it might be a good solution instead of BiocManager (I'll try to test that).

Reply:
ah ok so it does something for remotes::install_ but not `install.packages()`, correct?

So there are actually 3 cases

* installation with `remotes` during review

* installation with `install.packages()` (if the package goes on CRAN)

* installation with `BiocManager` if needed

Reply:
Sorry I probably wasn't clear enough. 

Where I have found `biocViews:` to work: When a package in being **installed from source** code, has a bioconductor package as a dependency and there is a compatible binary available for said dependency on bioconductor.

Where I have found `biocViews:` NOT to work: when the bioconductor dependency itself needs installing from source (ie no compatible binary found on bioconductor).

Reply:
In the issue you mentioned @annakrystalli Jim Hester said "You need to add a BiocViews field to your DESCRIPTION file, this indicates your package is a Bioconductor package " so that's not something about having Bioconductor dependencies, it's for (future) Bioconductor packages?

Reply:
Indeed, but from scouring the internet, I think it also works for indicating bioconductor dependencies, although it seems a bit of a _trick/hack_.

See for example this [comment](https://github.com/r-lib/devtools/issues/700#issuecomment-235127291) from where I think the tip might have started propagating. 

There's then more examples of folks trying to figure out how and why it works in the wild, eg [here](https://stackoverflow.com/questions/54411495/bioconductor-packages-not-installing-with-biocviews-specification-in-description).

Given all this confusion and hackery, perhaps we should invite someone more formally from Bioconductor to weigh in on our guidance?